File: algorithms/easy/minimum_difference_between_highest_and_lowest_of_k_scores.rs
impl Solution {
    pub fn minimum_difference(mut nums: Vec<i32>, k: i32) -> i32 {
        nums.sort();

        let k = k as usize;
        if nums.len() <= k {
            return nums[nums.len() - 1] - nums[0];
        }

        let mut ans = i32::MAX;
        for i in k - 1..nums.len() {
            ans = ans.min(nums[i] - nums[i + 1 - k]);
        }

        ans
    }
}

File: algorithms/easy/minimum_difference_between_highest_and_lowest_of_k_scores.kt
class Solution {
  fun minimumDifference(nums: IntArray, k: Int): Int {
    nums.sort()

    var ans = Int.MAX_VALUE
    for (i in (k - 1)..<nums.size) {
      ans = min(ans, nums[i] - nums[i - k + 1])
    }

    return ans
  }
}
